circle.yml

```diff
@@ -0,0 +1,3 @@
+machine:
+  node:
+    version: 5.12.0
```

Comment:
Hey @bkowshik 👋 Sorry for replying late. So this is kind of circleci + node + ubuntu version deal. 
So as per  https://circleci.com/gh/mapbox/osm-compare/edit#build-environment, we are using ubuntu 12.04
And it supports discrete versions for node which are listed here. https://circleci.com/docs/1.0/build-image-precise/#nodejs
And for now we are using the latest version supported by 12.04 ubuntu i.e. `5.5.0`

>What functionality needs node version 5.12.0?

  When we don't have any circle.yml configuration set, we get issue with [vt2geojson](https://github.com/mapbox/vt2geojson)
`TypeError: zlib has no method 'gunzipSync'` 

> What happens when consumers of this module use node version other than 5.12.0. Ex: 4.x or 6.x?

The error which is coming with vt2geojson is resolved with node version `v0.11.12`. [Ref](https://nodejs.org/dist/latest-v7.x/docs/api/zlib.html#zlib_zlib_gunzipsync_buf_options) . By default circleci uses node version `0.10.33` which is mentioned [here](https://circleci.com/docs/1.0/build-image-precise/#nodejs).